src/isomorphic/children/sliceChildren.js

```diff
@@ -32,16 +32,16 @@ function sliceChildren(children, start, end) {
   var slicedChildren = {};
   var flattenedMap = flattenChildren(children);
   var ii = 0;
+  var endIsDefined = end != null;
   for (var key in flattenedMap) {
     if (!flattenedMap.hasOwnProperty(key)) {
       continue;
     }
-    var child = flattenedMap[key];
     if (ii >= start) {
-      slicedChildren[key] = child;
+      slicedChildren[key] = flattenedMap[key];
     }
     ii++;
-    if (end != null && ii >= end) {
```

Comment:
This is really a micro optimization and seems like overkill (to me at least), *if* this is something we want to do I'd probably leave it at just `if (end == null) end = 0x7FFFFFFF;` (or w/e) and avoid the `end`-test inside the loop.